### PR TITLE
[TACHYON-1105] Added JSPC compiler to precompile JSP files

### PR DESCRIPTION
**Pull request by yupeng9**
https://tachyon.atlassian.net/browse/TACHYON-1105

So that JSP errors can be found at compile time

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -390,7 +390,7 @@
         <configuration>
           <configLocation>${checkstyle.path}tachyon_checks.xml</configLocation>
           <includeTestSourceDirectory>true</includeTestSourceDirectory>
-          <excludes>**/tachyon/thrift/**</excludes>
+          <excludes>**/org/apache/jsp/**,**/tachyon/thrift/**</excludes>
```
Comment (gpang) [inline pom.xml]: Is this the correct way to specify multiple excludes? I've seen other examples in pom files that list a new <exclude></exclude> per entry, but I don't know if that paradigm applies here.

**Comment by gpang** [comment]
This sounds good. Is there any negative side effect with pre-compiling the jsp files?

File: pom.xml (hunk 1 above)
Comment (yupeng9) [inline pom.xml]: Yes, it is the way: http://stackoverflow.com/questions/2362652/excluding-classes-in-maven-checkstyle-plugin-reports

**Comment by yupeng9** [comment]
For this one, I cannot think of any downside, because we just compile the jsp files for type checking without using its output.

**Comment by gpang** [comment]
LGTM

**Comment by calvinjia** [comment]
@yupeng9 I'm reverting this since its breaking the web ui.

**Pull request merged**